src/types/Transaction.ts

```diff
@@ -63,6 +62,111 @@ export enum TransactionVersion {
   Deploy
 }
 
+/**
+ * Represents a transaction hash, which can either be associated with a `Deploy` or a `TransactionV1`.
+ * This class helps in wrapping transaction hashes from different transaction types.
+ */
+@jsonObject
+export class TransactionHash implements IHash {
+  /**
+   * The hash associated with the deploy transaction, if applicable.
+   * This will contain the hash of the `Deploy` transaction.
+   */
+  @jsonMember({
+    name: 'Deploy',
+    constructor: Hash,
+    deserializer: json => {
+      if (!json) return;
+      return Hash.fromJSON(json);
+    },
+    serializer: value => {
+      if (!value) return;
+      return value.toJSON();
+    }
+  })
+  public deploy?: Hash;
+
+  /**
+   * The hash associated with the version 1 transaction, if applicable.
+   * This will contain the hash of the `TransactionV1`.
+   */
+  @jsonMember({
+    name: 'Version1',
+    constructor: Hash,
+    deserializer: json => {
+      if (!json) return;
+      return Hash.fromJSON(json);
+    },
+    serializer: value => {
+      if (!value) return;
+      return value.toJSON();
+    }
+  })
+  public transactionV1?: Hash;
+
+  /**
+   * Constructs a new `TransactionHash` instance, which can hold either a `Deploy` hash or a `TransactionV1` hash.
+   * @param deploy The hash of the deploy transaction, if applicable.
+   * @param transactionV1 The hash of the version 1 transaction, if applicable.
+   */
+  constructor(deploy?: Hash, transactionV1?: Hash) {
+    this.deploy = deploy;
+    this.transactionV1 = transactionV1;
```

Comment:
Maybe let's make the constructor private and add two factory methods:
- `fromDeployHash`
- `fromTransactionHash`